### PR TITLE
[REVIEW] CSV Reader: Use column wrapper for gdf_column output alloc/dealloc

### LINKED ISSUE #1318: [FEA] Move column/device memory wrapper utilities out of Parquet reader
**Is your feature request related to a problem? Please describe.**
The cuDF Parquet reader uses some helper wrapper utilities for alloc/dealloc. These classes provide RAII to the current C-like structs and interact better with using exceptions. We have the CSV Reader take advantage of them.

**Describe the solution you'd like**
Move the wrapper classes out of Parquet reader and into a utils header. Refactor the CSV reader use them.

**Describe alternatives you've considered**
Eventually, gdf_column will be replaced with a proper C++ class, and we can re-visit to remove the wrapper.

**Additional context**
See Parquet reader for how to use the wrapper classes. CSV reader should be similar.

### PR DESCRIPTION
**Pull request by j-ieong**
Fixes #1318.

* Use the gdf_column wrapper like Parquet reader
* Move the wrapper classes to an IO wrapper utils file
* Remove separate string_pair memory alloc and use column->data directly for intermediate parsing

### CODE REVIEW

File: cpp/src/io/csv/csv_reader.cu [hunk 1]
```diff
@@ -386,17 +391,17 @@ gdf_error setRecordStarts(const char *h_data, size_t h_size, raw_csv_t *raw_csv)
...
  *
- * @return gdf_error
+ * @return gdf_error GDF_SUCCESS if successful, otherwise an error code.
  *---------------------------------------------------------------------------**/
 gdf_error read_csv(csv_read_arg *args)
 {
-	gdf_error error = gdf_error::GDF_SUCCESS;
+  gdf_error error = gdf_error::GDF_SUCCESS;
+
+  std::vector<gdf_column_wrapper> columns;
```
Comment (vuule) [inline cpp/src/io/csv/csv_reader.cu]: I would prefer this to be declared just before it's used.

File: cpp/src/io/utilities/wrapper_utils.hpp [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    col = nullptr;
+    return temp;
+  }
+
+ private:
+  gdf_column *col = nullptr;
+};
+
+/**
+ * @brief A helper class that wraps fixed-length device memory for the GPU, and
```
Comment (vuule) [inline cpp/src/io/utilities/wrapper_utils.hpp]: Just to confirm - there is no automatic synchronization between host and device buffers?

File: cpp/src/io/csv/csv_reader.cu (hunk 1 above)
Comment (j-ieong) [reply cpp/src/io/csv/csv_reader.cu]: Done, thanks!

File: cpp/src/io/utilities/wrapper_utils.hpp (hunk 2 above)
Comment (j-ieong) [reply cpp/src/io/utilities/wrapper_utils.hpp]: Correct. For the first iteration, I've just kept the wrapper very simple and had the user perform any explicit synchronization required. This made it easier to port over the original Parquet reader code during development.
We can certainly add implicit sync, or helper methods to sync though.

**Pull request merged**